### PR TITLE
storcon: return an error for drain attempts while paused

### PR DESCRIPTION
**Pull request by VladLazar**
## Problem

We currently allow drain operations to proceed while the node policy is paused.

## Summary of changes

Return a precondition failed error in such cases. The orchestrator is updated in https://github.com/neondatabase/infra/pull/2544 to skip drain and fills if the pageserver is paused.

Closes: https://github.com/neondatabase/neon/issues/9907

### CODE REVIEW

**Review by jcsp — APPROVED** [review]
LGTM - we need to merge this one after the infra change I guess?

**Comment by VladLazar** [comment]
> LGTM - we need to merge this one after the infra change I guess?

Yes, this can merge at any point after the infra change merges. No prod release coordination required.

**Pull request merged**
